syntax = "proto3";

package Whiteboard.Drawing;

import "drawing_types.proto";

service DrawingService {
  rpc DrawingStream (stream DrawingEvent) returns (stream DrawingEvent);
}

message StreamEvent {
  string client_id = 1;
  oneof EventType {
    DrawingEvent drawingEvent = 2;
    EraseEvent eraseEvent = 3;
  }
}

message DrawingEvent {
  Whiteboard.Types.Drawable drawable = 1;
}

message EraseEvent {
  Whiteboard.Types.Drawable drawable = 1;
}
